src/Runner.Worker/ActionManager.cs

```diff
@@ -1068,8 +1233,11 @@ private ActionContainer PrepareRepositoryActionAsync(IExecutionContext execution
                 }
                 else if (actionDefinitionData.Execution.ExecutionType == ActionExecutionType.Composite)
                 {
-                    Trace.Info($"Action composite: {(actionDefinitionData.Execution as CompositeActionExecutionData).Steps}, no more preparation.");
-                    return null;
+                    // TODO: we need to generate unique Id's for composite steps
```

Comment:
new guid should be fine

Steps inside composite share the same timeline record id as their parent (ExecutionContext.CreateEmbeddedChild). Regular step ID (guid) matches the execution context ID and timeline record ID, iirc. I might be wrong, but that is my best guess because that's why those concepts exist.

So the new guid here is probably just needed to manage building/starting the image.